core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,896 @@ def update_question_suggestion(
+            stat.submitted_translation_word_count)
+        stats_model.accepted_translations_count = (
+            stat.accepted_translations_count)
+        stats_model.accepted_translations_without_reviewer_edits_count = (
+            stat.accepted_translations_without_reviewer_edits_count)
+        stats_model.accepted_translation_word_count = (
+            stat.accepted_translation_word_count)
+        stats_model.rejected_translations_count = (
+            stat.rejected_translations_count)
+        stats_model.rejected_translation_word_count = (
+            stat.rejected_translation_word_count)
+        stats_model.contribution_dates = stat.contribution_dates
+
+    assert stats_models_to_update is not None
+    suggestion_models.TranslationContributionStatsModel.update_timestamps_multi(
+        stats_models_to_update,
+        update_last_updated_time=True)
+    suggestion_models.TranslationContributionStatsModel.put_multi(
+        stats_models_to_update)
+
+
+def _update_translation_review_stats_models(
+    translation_review_stats: List[
+        suggestion_registry.TranslationReviewStats]
+) -> None:
+    """Updates TranslationReviewStatsModel models for given translation
+    review stats.
+
+    Args:
+        translation_review_stats: list(TranslationReviewStats). A list of
+            TranslationReviewStats domain objects.
+    """
+    stats_dict = {}
+
+    for stat in translation_review_stats:
+        stat_id = suggestion_models.TranslationReviewStatsModel.construct_id(
+            stat.language_code, stat.contributor_user_id, stat.topic_id)
+        stats_dict[stat_id] = stat
+
+    stats_ids = stats_dict.keys()
+
+    stats_models = (
+        suggestion_models.TranslationReviewStatsModel.get_multi(list(
+            stats_ids)))
+
+    stats_models_to_update: List[
+        suggestion_models.TranslationReviewStatsModel] = []
+    for stats_model in stats_models:
+        # We can confirm that stats_model will not be None since the for loop
+        # already gives the not None stats_model object. Hence we can rule out
+        # the possibility of None for mypy type checking.
+        assert stats_model is not None
+        stat = stats_dict[stats_model.id]
+        stats_model.reviewed_translations_count = (
+            stat.reviewed_translations_count)
+        stats_model.reviewed_translation_word_count = (
+            stat.reviewed_translation_word_count)
+        stats_model.accepted_translations_count = (
+            stat.accepted_translations_count)
+        stats_model.accepted_translations_with_reviewer_edits_count = (
+            stat.accepted_translations_with_reviewer_edits_count)
+        stats_model.first_contribution_date = (
+            stat.first_contribution_date)
+        stats_model.last_contribution_date = (
+            stat.last_contribution_date)
+
+    suggestion_models.TranslationReviewStatsModel.update_timestamps_multi(
+        stats_models_to_update,
```

Comment:
Same comment as above why are we providing an empty list here -- basically, we are not appending anything to this list. In line 2104 we are defining an empty list and here we are passing that list but in between, we are not appending anything? -- again please correct me If I'm wrong

Reply:
Done